src/Disks/IO/ThreadPoolRemoteFSReader.cpp

```diff
@@ -100,8 +100,10 @@ IAsynchronousReader::Result ThreadPoolRemoteFSReader::execute(Request request)
     IAsynchronousReader::Result read_result;
     if (result)
     {
-        read_result.size = reader.buffer().size();
-        read_result.offset = reader.offset();
+        chassert(reader.buffer().begin() >= request.buf);
+        chassert(reader.buffer().end() <= request.buf + request.size);
+        read_result.offset = reader.position() - request.buf;
+        read_result.size = reader.buffer().end() - request.buf;
```

Comment:
Why do we need to do it this way now? With `-request.buf` I mean. Should it not be guaranteed that `request.buf == reader.buffer().begin()`?